lib/statusbar-summary-view.coffee

```diff
@@ -23,20 +24,28 @@ class StatusBarSummaryView
 
     return if info + warning + error == 0
 
-    el = new StatusBarSummary(info, warning, error)
+    el = new StatusBarSummary(messages, editor, info, warning, error)
     @tile = statusBar.addRightTile({item: el, priority: 100})
 
 class StatusBarSummary extends View
-  @content: (info, warning, error) ->
+  initialize: (messages, editor) ->
+    @on 'click', '.linter-summary-click-container', ->
+      moveToNextMessage messages, editor
+
+  @content: (messages, editor, info, warning, error) ->
     @div class: 'linter-summary inline-block', =>
-      if error > 0
-        @div class: 'linter-error inline-block', error, =>
-          @span class: 'icon-right'
-      if warning > 0
-        @div class: 'linter-warning inline-block', warning, =>
-          @span class: 'icon-right'
-      if info > 0
-        @div class: 'linter-info inline-block', info, =>
-          @span class: 'icon-right'
+      @div class: 'linter-summary-click-container', =>
```

Comment:
out of curiosity, why the extra div? did attaching the event handler to `.linter-summary` not work?